### PR TITLE
bpo-39879: update language reference to specify that dict is insertion-ordered.

### PR DESCRIPTION
**Pull request by furkanonder**
https://bugs.python.org/issue39879

### CODE REVIEW

**Review by hakancelikdev — APPROVED** [review]

**Review by hakancelikdev — APPROVED** [review]

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -436,6 +436,11 @@ Mappings
       additional examples of mapping types, as does the :mod:`collections`
       module.
 
+      .. versionchanged:: 3.7
+         Dictionary order is guaranteed to be insertion order. Note that updating a key
```
Comment (isidentical) [inline Doc/reference/datamodel.rst]: I feel like something is missing here, please ignore if this is the common usage (I'm not a native speaker). 
```suggestion
         The dictionary order is guaranteed to be the same as the insertion order. Note that updating a key
```

File: Doc/reference/datamodel.rst [hunk 2]
```diff
@@ -436,6 +436,11 @@ Mappings
...
+         does not affect the order. Keys added after deletion are inserted at the end.
+         This behavior was an implementation detail of CPython from 3.6.
```
Comment (isidentical) [inline Doc/reference/datamodel.rst]: I think you should split this into its own version changed tag.

**Review by isidentical — COMMENTED** [review]
Looks great

File: Doc/reference/datamodel.rst [hunk 3]
```diff
@@ -436,6 +436,11 @@ Mappings
       additional examples of mapping types, as does the :mod:`collections`
       module.
 
+      .. versionchanged:: 3.7
+         The dictionary order is guaranteed to be the same as the insertion order. Note that updating a key
```
Comment (zware) [inline Doc/reference/datamodel.rst]: Please wrap lines at 80 characters, as [mentioned in the devguide](https://devguide.python.org/documenting/#use-of-whitespace).

File: Doc/reference/datamodel.rst (hunk 2 above)
Comment (zware) [reply Doc/reference/datamodel.rst]: I think this is fine being just a note in this block.

**Review by zware — CHANGES_REQUESTED** [review]
See also my comment on the issue.

File: Doc/reference/datamodel.rst (hunk 3 above)
Comment (furkanonder) [reply Doc/reference/datamodel.rst]: It's fixed.

**Comment by furkanonder** [comment]
It was combined with another pr. =>  https://github.com/python/cpython/pull/19006

**Pull request closed without merging**